src/compile/mark/init.ts

```diff
@@ -22,7 +22,10 @@ export function normalizeMarkDef(mark: Mark | MarkDef, encoding: Encoding<string
   // set opacity and filled if not specified in mark config
   const specifiedOpacity = getFirstDefined(markDef.opacity, getMarkConfig('opacity', markDef, config));
   if (specifiedOpacity === undefined) {
-    markDef.opacity = opacity(markDef.type, encoding);
+    markDef.opacity = opacity(markDef, encoding);
+  } else if (specifiedOpacity === 1.0) {
+    // Vega's default opacity is 1.0 so it's safe to delete the opacity in this case
+    delete markDef.opacity;
```

Comment:
What if the user sets `markDef.opacity` to 1 and the config is 0.5? Wouldn't this break it?

Reply:
Yeah, you're right. I didn't think of that case.

Reply:
That's why they pay me the big bucks 💰 (not really, actually...)

Reply:
Hmm, the first fix that comes to my mind would be to set the mark's opacity to the first defined of config opacity or 1.0 during transform extract and then delete the mark's opacity in `normalizeMarkDef(...)` if the mark's opacity is equal to the first defined of config opacity or 1.0.

This solution does seem pretty dirty though.

Reply:
Actually, I think I was wrong. The mark config we read from here is a Vega-Lite specific config, I think. Can you try the case I described above and delete the opacity from the Vega spec manually? I believe it should not change anything.

Reply:
Remind me why we need to apply the config property to the mark then? Isn't that counter the purpose of a config? @kanitw

Reply:
If I remember correctly, we set default config for opacity to be `undefined`, so that by default we use our smart rules.  When user set the config.mark.opacity, then we use that exact opacity number.

Reply:
👍 makes sense now.